src/main/webapp/app/core/admin/standardized-competencies/import/admin-import-standardized-competencies.component.ts

```
@@ -196,9 +207,100 @@ export class AdminImportStandardizedCompetenciesComponent {
+            }
+        }
+    }
+
+    /**
+     * Validates the parsed import data against server-side constraints.
+     * Returns an array of i18n-translated error strings; empty array means valid.
+     */
+    private validateImportData(data: KnowledgeAreasForImportDTO): string[] {
+        const errors: string[] = [];
+        const sourceIds = new Set((data.sources ?? []).map((s) => s.id).filter((id): id is number => id !== undefined));
+        (data.sources ?? []).forEach((source, index) => this.validateSource(source, index + 1, errors));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)